src/main/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImpl.java

```diff
@@ -0,0 +1,371 @@
+
+        return data;
+    }
+
+    /**
+     * Get the Link containing the ixbrl location and the relationship link e.g. accounts.
+     *
+     * @param ixbrlLocation - the location where ixbrl is stored
+     * @return {@link List<Link>}
+     */
+    private List<Link> getFilingLinks(String ixbrlLocation) {
+        Link link = new Link();
+        link.setRelationship(LINK_RELATIONSHIP);
+        link.setHref(ixbrlLocation);
+
+        return Arrays.asList(link);
+    }
+
+    /**
+     * Get the description values, which currently contains the period end date. This data is
+     * required for the filing description.
+     *
+     * @param periodEndDate - account's period end date
+     * @return {@link Map<String,String>} containing period end date, to match filing model
+     */
+    private Map<String, String> getDescriptionValues(LocalDate periodEndDate) {
+        Map<String, String> descriptionValues = new HashMap<>();
+        descriptionValues.put(PERIOD_END_ON, periodEndDate.toString());
+
+        return descriptionValues;
+    }
+
+    /**
+     * Get the description for the filing. The description for the account type is retrieved by
+     * using DocumentDescriptionHelper class.
+     *
+     * @param accountsType - Account type information: account type, ixbrl's template name, account
+     * @param periodEndDate - account's period end date
+     * @return accounts description.
+     * @throws IOException
+     */
+    private String getFilingDescription(AccountsType accountsType, LocalDate periodEndDate)
+        throws IOException {
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put(PERIOD_END_ON, periodEndDate);
+
+        return DocumentDescriptionHelper
+            .getDescription(accountsType.getFilingDescriptionKey(), parameters);
+    }
+
+    /**
+     * Validates the ixbrl against TNEP. This validation is driven by the env. variable and it can
+     * be disable.
+     */
+    private boolean isValidIXBL() {
+
+        boolean isIxbrlValid = true;
+        if ("true".equals(getMandatoryEnvVariable(DISABLE_IXBRL_VALIDATION_ENV_VAR))) {
+            //TODO TNEP validation needs to be added. Copy logic from abridged. Next PR
+        }
+
+        return isIxbrlValid;
+    }
+
+    /**
+     * Generates the ixbrl by calling the document render service.
+     *
+     * @param accountsType - Account type information: account type, ixbrl's template name, account
+     * @param requestBody - the http request request (json format)
+     * @return The location where the service has stored the generated ixbrl.
+     */
+    private String callIxbrlGenerator(AccountsType accountsType, String requestBody)
+        throws IOException {
+        if (requestBody != null) {
+            DocumentGeneratorConnection connection = getDocumentGeneratorConnection(accountsType,
+                requestBody);
+            return ixbrlGenerator.generateIXBRL(connection);
+        }
+
+        return null;
```

Comment:
Can you add logging where you return null please

Reply:
4a47945